notebooks/working/investigating_deconfounder.py

```diff
@@ -114,35 +114,54 @@
 # -
 
 # ## Specifiy the paramteric relationship between the covariates. 
+#
+# This is where we specify the relationship between the variables in the Causal graph above. Specifically, we have an equation that relates each two connected nodes on the graph (so an equation for each edge).
+#
+# In the outcome model for y, the coefficient on each of the variables a, b, c, d, e and f represents it's $true$ causal effect. The coefficient on the confounder, however, does not represent its causal effect, because it has 5 decendents that are part of the outcome model. 
+#
+# We are not ultimately interested in the causal effect of the confounder for this exercice, and instead are interested in the coefficients on each of its descendents. 
 
 # +
 sample_size = 10000
 
 confounder = np.random.normal(loc=20, scale = 10, size = sample_size)
+# -
 
-# +
+## Confounding structure
 a = .3*confounder + np.random.normal(loc=10,scale=3,size=sample_size)
 b = -1*confounder + np.random.normal(loc=6,scale=3,size=sample_size)
 c = np.random.normal(loc=15,scale=3,size=sample_size) # + .5*confounder 
 d = .8*confounder + np.random.normal(loc=-10,scale=3,size=sample_size)
 e = .5*confounder + np.random.normal(loc=8,scale=3,size=sample_size)
-f= -.3*confounder + np.random.normal(loc=-12,scale=3,size=sample_size)
+f = -.3*confounder + np.random.normal(loc=-12,scale=3,size=sample_size)
 
+# +
+## Generating coefficients
+beta_a = np.random.normal(loc=2, scale=1, size=sample_size)
+beta_b = np.random.normal(loc=2, scale=1, size=sample_size)
+beta_c = np.random.normal(loc=-5, scale=1, size=sample_size)
+beta_d = np.random.normal(loc=3, scale=1, size=sample_size)
+beta_e = np.random.normal(loc=2, scale=1, size=sample_size)
+beta_f = np.random.normal(loc=-5, scale=1, size=sample_size)
```

Comment:
Maybe not necessary, but you can have one dictionary storing the mean and std of each of these parameters.